### PR TITLE
Week3 Tasks

### PR DESCRIPTION
**Pull request by Vanshbit138**
Inventory Manager – Week 3

### Summary
This PR adds the `InventoryManager` class to load, validate, and manage products from a CSV file using Pydantic models.

### Context
As part of the OOP refactoring task, we needed to encapsulate all inventory-related operations in a dedicated class to adhere to SRP and improve code organization.

### Changes Made
- Created `InventoryManager` class in `inventory.py`
- Implemented CSV loading and product classification logic
- Integrated logging for invalid product entries
- Generated a low-stock report using utility functions

### Testing
- Loaded sample CSVs and verified product parsing
- Verified low-stock report generation
- All methods tested with valid and invalid data inputs

### Additional Notes
- Pydantic is used for validation, and fallback logging is handled via the `log_error` utility

## Sample Output
Below is a sample output of the Inventory Report generated after running the system:
image

### Day 1 – Week 4 Tasks Covered
✅ Set up pytest for the project.

✅ Created unit tests for all product models.

✅ Covered validation edge cases (zero price, negative quantity, etc.).

✅ Documented test results and model validations.

## Sample Output
Below is a sample output generated after running the system:
image

### CODE REVIEW

**Comment by decode-mayank** [comment]
Please add a proper description in the 'Description' field explaining what this PR does

**Comment by decode-mayank** [comment]
Add SS for the output of Dashboard

**Comment by decode-mayank** [comment]
add label

File: Week-3/inventory_manager/models.py [hunk 1]
```diff
@@ -0,0 +1,33 @@
+from pydantic import BaseModel, Field, ConfigDict
+from typing import Optional
+from datetime import date
+
+
+class Product(BaseModel):
```
[the remark this answers is not in the document]
Comment (Vanshbit138) [reply Week-3/inventory_manager/models.py:6]: Done Sir!

File: README.md [hunk 2]
```diff
@@ -134,4 +154,56 @@ git add .
...
+│   └── utils.py                  # Logging and reporting utilities
+│
+├── venv/                         # (Optional) Virtual environment for dependencies
+│
+├── main.py                       # Entry point script to run inventory system│
+├── pyproject.toml                # Config for black, ruff, etc.
+├── requirements.txt              # Python package dependencies
+```
+
+## ▶️ How to Run the Inventory Manager (Week 3)
```
[the remark this answers is not in the document]
Comment (Vanshbit138) [reply README.md]: Done sir!

**Review by decode-mayank — APPROVED** [review]

**Review by sanidhyabitcot — APPROVED** [review]
LGTM
